superset/commands/dataset/template_columns/get.py

```diff
@@ -0,0 +1,74 @@
+from functools import partial
+from typing import Any, List, Optional
+
+from superset.commands.base import BaseCommand
+from superset.commands.dataset.exceptions import DatasetNotFoundError
+from superset.commands.dataset.template_columns.exceptions import (
+    DatasetGetTemplateColumnsError,
+)
+from superset.connectors.sqla.models import SqlaTable
+from superset.daos.dataset import DatasetDAO
+from superset.jinja_context import JinjaTemplateProcessor, safe_proxy
+from superset.utils.core import DatasourceType
+from superset.utils.decorators import on_error, transaction
+
+
+class TemplateColumnReferencesProcessor(JinjaTemplateProcessor):
+    """For extracting column names from a SQL query. Replace the Jinja functions
+    with custom ones that only extract column names from the query."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._columns = set()
+
+    def url_param(self, param: str, **kwargs) -> Optional[str]:
+        self._columns.add(param)
+        return None
+
+    def filter_values(self, column: str, **kwargs) -> list[Any]:
+        self._columns.add(column)
+        return []
+
+    def get_filters(self, column: str, remove_filter: bool = False) -> list[Any]:
+        self._columns.add(column)
+        return []
```

Comment:
why the two function have same implementation, and why it returns empty []? btw, could this be a `@property`?